genienlp/data_utils/iterator.py

```diff
@@ -53,9 +53,9 @@ def __init__(self, data_source, batch_size, sort, shuffle_and_repeat, sort_key_f
         self.groups = groups
         
         if sort:
-            self.data_source = list(sorted(data_source, key=self.sort_key))
+            self.data_source, self.original_order = tuple(zip(*sorted(list(zip(data_source, range(len(data_source)))), key=lambda x: self.sort_key(x[0]))))
```

Comment:
This line is not very readable. Perhaps a comment explaining what all the zips and lists and what not do? Or even better, rewrite it in multiple clean imperative lines...